examples/hosts/app-integration/schema-upgrade/src/appView.tsx

```diff
@@ -0,0 +1,110 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React, { useEffect, useRef, useState } from "react";
+
+import { IContainerKillBit } from "./containerKillBit";
+import { IInventoryList } from "./inventoryList";
+import { InventoryListView } from "./inventoryView";
+
+export interface IAppViewProps {
+    inventoryList: IInventoryList;
+    // Normally there's no need to display the imported string data, this is for demo purposes only.
+    importedStringData: string | undefined;
+    // Normally this is probably a Promise<void>.  Returns a string here for demo purposes only.
+    writeToExternalStorage: () => Promise<string>;
+    saveAndEndSession: () => Promise<void>;
+    containerKillBit: IContainerKillBit;
+}
+
+export const AppView: React.FC<IAppViewProps> = (props: IAppViewProps) => {
+    const {
+        inventoryList,
+        importedStringData,
+        writeToExternalStorage,
+        containerKillBit,
+        saveAndEndSession,
+    } = props;
+
+    const [dead, setDead] = useState<boolean>(containerKillBit.dead);
+    const [sessionEnding, setSessionEnding] = useState<boolean>(containerKillBit.markedForDestruction);
+
+    useEffect(() => {
+        const deadHandler = () => {
+            setDead(containerKillBit.dead);
+        };
+        containerKillBit.on("dead", deadHandler);
+        // For some reason, I'm seeing the event fire between setting the state initially and adding the listener.
+        deadHandler();
```

Comment:
Hm that is strange, you are not yielding between setting initial state and here.

Reply:
I'll need to look into this more, I haven't examined it since last ~August or so :-P